File: problems/arrays/q2/Q2.php
<?php
// Given an array, 
// rotate the array to the right by k steps, 
// where k is non-negative.

function rotateArrayMethod1($array = [], $k) {
    if (count($array) == $k OR count($array) == 0) {
        return $array;
    }

    $counter = 1;
    for ($i = 0; $i < count($array); $i++) {
        if ($counter <= $k) {
            $lastItem = array_pop($array);
            array_unshift($array, $lastItem);
            $counter++;
        }
    }

    return $array;
}

$array = [1, 2, 3, 4, 5];
$k = 3;

print_r(rotateArrayMethod1($array, $k));

File: problems/arrays/q2/Q2.R
# Given an array, 
# rotate the array to the right by k steps, 
# where k is non-negative.

rotate_array_method_1 <- function(array, k) {
    if (length(array) == k || length(array) == 0) {
        return(array)
    }

    counter = 1
    for (i in seq_along(array)) {
        if (counter <= k) {
            last_item = array[length(array)] # get the last item
            array = array[-length(array)] # delete the last item
            array = c(last_item, array) # add the last item
            counter = counter + 1
        }
    }

    return(array)
}

array = c(1,2,3,4,5)
k = 3

rotate_array_method_1(array, k)